tests/cnf/core/network/metallb/tests/common.go

```diff
@@ -156,6 +158,25 @@ func createBGPPeerAndVerifyIfItsReady(
 	}
 }
 
+func setupBgpAdvertisementAndIPAddressPool(addressPool []string, prefixLen int32) *metallb.IPAddressPoolBuilder {
+	ipAddressPool, err := metallb.NewIPAddressPoolBuilder(
+		APIClient,
+		"addresspool",
+		NetConfig.MlbOperatorNamespace,
+		[]string{fmt.Sprintf("%s-%s", addressPool[0], addressPool[1])}).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create IPAddressPool")
+
+	_, err = metallb.
+		NewBGPAdvertisementBuilder(APIClient, "bgpadvertisement", NetConfig.MlbOperatorNamespace).
+		WithIPAddressPools([]string{ipAddressPool.Definition.Name}).
+		WithCommunities([]string{"65535:65282"}).
+		WithLocalPref(100).
+		WithAggregationLength4(prefixLen).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create BGPAdvertisement")
+
+	return ipAddressPool
+}
+
```

Comment:
IMO better convert int to int32 on this level
```
func setupBgpAdvertisementAndIPAddressPool(addressPool []string, prefixLen int) 
...
		WithAggregationLength4(int32(prefixLen)).Create()


```

Reply:
updated